src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java

```diff
@@ -275,18 +275,11 @@ public MethodDefinition build() {
         }
 
         if (method.returnExpr() != null && !isLastStatementThrowExpr) {
-          if (method.returnType().isPrimitiveType()) {
-            Preconditions.checkState(
-                method.returnExpr().type().isPrimitiveType()
-                    && method.returnType().equals((method.returnExpr().type())),
-                "Method primitive return type does not match the return expression type");
-
-          } else {
-            Preconditions.checkState(
-                !method.returnExpr().type().isPrimitiveType()
-                    && method.returnType().isSupertypeOrEquals(method.returnExpr().type()),
-                "Method reference return type is not a subtype of the return expression type");
-          }
+          TypeNode returnExprType = method.returnExpr().type();
```

Comment:
What's the motivation for this change? Keeping the checks separate makes them clearer, though we can remove the redundant `!isPrimitiveType()` check in the else-block above.

Reply:
Cool, kept the original checks to make them clearer. 
1. removed redundant `isPrimitiveType()` in the first block, since returnExpr type can be reference type for boxed/primitive type equality.
2. removed `isPrimitiveType()` check in else block, but `isSupertypeOrEquals ` only checks when both types are reference types. So added `equals()` condition to allow boxed/primitive types.